include/swift/SIL/SILConstants.h

```diff
@@ -385,6 +387,36 @@ inline llvm::raw_ostream &operator<<(llvm::raw_ostream &os, SymbolicValue val) {
   return os;
 }
 
+/// This is a representation of a memory object referred to be an address.
+/// Memory objects may be mutated over their lifetime, but their overall type
+/// remains the same.
+struct SymbolicValueMemoryObject {
+  Type getType() const { return type; }
+
+  SymbolicValue getValue() const { return value; }
+  void setValue(SymbolicValue newValue) { value = newValue; }
+
+  /// Create a new memory object whose overall type is as specified.
+  static SymbolicValueMemoryObject *create(Type type,
+                                           llvm::BumpPtrAllocator &allocator);
+  static SymbolicValueMemoryObject *create(Type type, SymbolicValue value,
```

Comment:
it'd be slightly more efficient to combine both create() into one, where value defaults to `SymbolicValue::getUninitMemory`. This way in the second form of `create()` (with a real `value`), we do not write to `SymbolicValueMemoryObject::value` twice.

Also, will we be able to remove the `setValue()` API in the case?

Reply:
Good idea, I'll do that in my next patch.

We need to retain setValue() because the model of memory is mutable.  We have stores etc against memory and it needs to be updatable.